src/Tribe/Context.php

```diff
@@ -91,41 +137,445 @@ public function is_editing_post( $post_or_type = null ) {
 	 * Helper function to indicate whether the current execution context is AJAX.
 	 *
 	 * This method exists to allow us test code that behaves differently depending on the execution
-	 * context; passing a value to this argument will set it to that value in future checks, a test-related usage.
+	 * context.
 	 *
 	 * @since 4.7.12
-	 *
-	 * @param bool $doing_ajax An injectable status to override the `DOING_AJAX` check.
+	 * @since TBD Removed the $doing_ajax parameter.
 	 *
 	 * @return boolean
 	 */
-	public function doing_ajax( $doing_ajax = null ) {
-		if ( null !== $doing_ajax ) {
-			$this->doing_ajax = (bool) $doing_ajax;
+	public function doing_ajax() {
+		return function_exists( 'wp_doing_ajax' )
+			? wp_doing_ajax()
+			: defined( 'DOING_AJAX' ) && DOING_AJAX;
+	}
+
+	/**
+	 * Checks whether the context of the current HTTP request is a Cron one or not.
+	 *
+	 * @since 4.7.23
+	 * @since TBD Removed the $doing_cron parameter.
+	 *
+	 * @return bool Whether the context of the current HTTP request is a Cron one or not.
+	 */
+	public function doing_cron() {
+		return function_exists( 'wp_doing_cron' )
+			? wp_doing_cron()
+			: defined( 'DOING_CRON' ) && DOING_CRON;
+	}
+
+	/**
+	 * Returns the current display mode.
+	 *
+	 * This is read from the the request variables first and from the main query object second.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The global event display or an empty string if not set in any of the sources above.
+	 */
+	public function get_event_display() {
+		$request_value = tribe_get_request_var( 'tribe_event_display' );
+		global $wp_query;
+		$query_value = $wp_query->get( 'eventDisplay', '' );
+
+		return $request_value ? $request_value : $query_value;
+	}
+
+	/**
+	 * Gets a value reading it from the location(s) defined in the `Tribe__Context::$props
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key The key of the variable to fetch.
+	 * @param mixed|null $default The default value to return if not found.
+	 *
+	 * @return mixed The value from the first location that can provide it or the default
+	 *               value if not found.
+	 */
+	public function get( $key, $default = null ) {
+		$value         = $default;
+		$the_locations = self::$locations[ $key ];
+
```

Comment:
Maybe we could also create a `tribe_context_pre_{$key}` to allow short-circuiting the whole thing for a particular purpose?